src/System.Device.Gpio.Tests/README.md

```diff
@@ -1,52 +1,127 @@
 - run `./build.sh` in the checkout directory. This will install the correct NET Core SDK in the .dotnet subfolder of the working copy.
 
 Build System.Device.Gpio.dll:
+
 ```
 cd src/System.Device.Gpio
 dotnet build System.Device.Gpio.sln
 ```
+
 This builds the main System.Device.Gpio assembly together with its test assembly. Before running the tests, you need to:
 - Connect BCM Pins 12 and 16 (physical Pins 32 and 36) with a cable. It is suggested to add a 1kΩ to 10kΩ resistor between the pins; this protects the Pi in the case of a misconfiguration (i.e both pins set to Out, one high, the other low). 
 - Leave BCM Pin 23 (physical Pin 16) open (not connected to anything).
 
 ## Raspberry Pi driver tests
+
 After that, you can run the tests with the RaspberryPiDriver (which is the default low-level driver for the Raspberry Pi) like:
+
 ```
 dotnet test --filter RaspberryPiDriverTests System.Device.Gpio.sln 
 ```
+
 Depending on the version of the Pi and the installed Linux distribution, it may be required to run the tests as root:
+
 ```
 sudo dotnet test --filter RaspberryPiDriverTests System.Device.Gpio.sln 
 ```
 
 If everything went smoothly, the output should end with a success message. 
 
+## Running tests depending on components/requirements
+
+Currently the full test suite requires following components, if you don't have one use following command line switches to skip
+
+| Requirements | Skip arguments |
+| --- | --- |
+| MCP 3008 (SPI)* | `-notrait feature=spi` |
+| BME280 (I2C) | `-notrait feature=i2c` |
+| Pins 5 and 6 connected | `-notrait feature=gpio` |
+| libgpiod | `-notrait feature=gpio-libgpiod` |
+| sysfs access (i.e. sudo or permissions) | `-notrait feature=gpio-sysfs` |
+| configured PWM (overlaps with MCP3008 setting) | `-notrait feature=pwm` |
+| root access (overlaps, assumes you use default permissions) | `-notrait requirement=root` |
+
+* Also inputs to ADC are required including one connected to PWM output through low pass filter, please refer to the schematics.
+
+
+
 ## SysFsDriver Tests
+
 You can also run the SysFsDriver Tests (this uses a more generic approach). This driver requires root permissions to run, so you need to `sudo` the command:
+
 ```
 sudo dotnet test --filter SysFsDriverTests System.Device.Gpio.sln 
 ```
 
 ## LibgpiodDriver Tests
+
 To run the Libgpiod Driver test (a Linux Kernel Driver for the GPIO device), you need to first install the libgpiod package:
+
 ```
 sudo apt install -y libgpiod-dev
 ```
+
 These tests do not require root permissions to run:
+
 ```
 dotnet test --filter LibGpiodDriverTests System.Device.Gpio.sln 
 ```
+
+## Full test run schematics
+
+Our CI is currently using [this schematics](board-schematics.pdf) with Raspberry PI 3.
+
+You can download [gerber files](board-gerber.zip) if you'd like to manufacture PCB yourself (this is a 2 layer board).
+
+Following list of components are needed for PCB:
+
+| Component | Quantity |
+| --- | --- |
+| MCP3008 I/SL | 1 |
```

Comment:
Apart from listing these requirements and showing our PCB, I think it would be useful if we also provided a fritzing diagram since most contributors will probably be using a breadboard to test this. You don't have to block this PR on that, but do at least log an issue so that we add this diagram so people know how to connect each of the components in order for our tests to run without a need to change the code configurations

Reply:
there is schematics pdf with description how to connect, I feel fritzing doesn't look any clearer and rather more confusing